lib/src/http_client.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+
+/// Exception thrown when an HTTP request fails
+class HttpException implements Exception {
+  final String message;
+
+  HttpException(this.message);
+
+  @override
+  String toString() => 'HttpException: $message';
+}
+
+/// Interface for HTTP client used by the Eppo API client
+abstract class EppoHttpClient {
```

Comment:
basic http client - nothing eppo specific. created it to be able to mock in tests.